crates/typst-pdf/src/embed.rs

```diff
@@ -0,0 +1,67 @@
+use crate::catalog::pdf_date;
+use crate::{PdfAConformanceLevel, PdfChunk, WithGlobalRefs};
+use ecow::EcoString;
+use pdf_writer::{Finish, Name, Ref, Str, TextStr};
+use std::collections::HashMap;
+use typst::diag::{bail, SourceResult};
+use typst::syntax::Span;
+
+pub fn write_embedded_files(
+    ctx: &WithGlobalRefs,
+) -> SourceResult<(PdfChunk, HashMap<EcoString, Ref>)> {
+    let mut chunk = PdfChunk::new();
+    let mut embedded_files = HashMap::default();
+    if !ctx.resources.embeds.is_empty()
+        && Some(PdfAConformanceLevel::A_2) == ctx.options.standards.pdfa
+    {
+        bail!(
+            Span::detached(),
+            "file embeddings are currently only supported for PDF/A-3"
```

Comment:
This would technically be allowed if the embedded file is PDF/A-2. However, as I didn't know how to check that here, I decided to restrict file embeddings to PDF/A-3.

Reply:
Shouldn't the message reflect that this is a limitation imposed by pdf/a-2, not typst?

Reply:
PDF/A-2 does allow file embeddings as long as the files are also PDF/A-2. So the restriction is part Typist and part PDF/A-2.

I am open for iterating on the message.